prow/apis/prowjobs/v1/types.go

```diff
@@ -1018,6 +1019,23 @@ func (j *ProwJob) ClusterAlias() string {
 	return j.Spec.Cluster
 }
 
+// NeedReportState return true if it need to report the new state for the job.
+func (j *ProwJob) NeedReportState(reporter string) bool {
+	if j.Status.PrevReportStates[reporter] != j.Status.State {
+		return true
+	}
+
+	if j.Spec.Agent != JenkinsAgent {
+		return false
+	}
```

Comment:
Is there a good reason why this behavior should be limited to Jenkins controller? If prow thinks the status should have a different description on the same state, then it should propagate no matter if the job is jenkins or anything else, no?

Reply:
If you agree, I cloud add it , but current only Jenkins agent has this case: 
```text
Prow job: triggered ------> pending ---------------------------------------------------> succeed or failed
                                                 \                                                                                                                       /
                                                  \                                                                                                                     /
                            Jenkins job: queued(no job url) --> running(job description and url changed) --> succeed or failed(only description changed)
```

Reply:
Perhaps a new state will make it simpler: "queuing", but other agents do not necessarily have the same state sets.

Reply:
Fair enough, I guess it's fine. I'm still curious about the main review comment.